starlette/websockets.py

```diff
@@ -144,13 +145,22 @@ async def send_json(self, data: typing.Any, mode: str = "text") -> None:
         else:
             await self.send({"type": "websocket.send", "bytes": text.encode("utf-8")})
 
-    async def close(self, code: int = 1000) -> None:
-        await self.send({"type": "websocket.close", "code": code})
+    async def close(self, code: int = 1000, reason: str = None) -> None:
+        if reason is None:
+            await self.send({"type": "websocket.close", "code": code})
+        else:
+            await self.send({"type": "websocket.close", "code": code, "reason": reason})
```

Comment:
That makes sense but it should be something like:

```python
websocket_close = WebSocketClose(code, reason)
await websocket_close(self.scope, self._receive, self._send)
```

This looks reasonable but a test is failing I'll take a look.

Reply:
What do you think about this?

Reply:
Actually when I first did this I thought that maybe we should get rid of `WebSocketClose`. I know it's out of scope and backwards incompatible.

But what you suggest is even better, it will make use of `WebSocketClose`, as I think it was meant to.
If you don't have any strong feelings about it I say let's wait for someone else's feedback here.